### ISSUE TITLE
Use structured logs 

### ISSUE THREAD

**Issue by yuanchen8911** [kubernetes-sigs/scheduler-plugins#120]
Logging in Kubernetes 1.19  has been migrated to the structured format. Two new methods are added to `k8s.io/klog/v2` library: `InfoS `and `ErrorS`. Should we update the scheduler plugins' logging calls to the structured logs and use the structured format in future logging? 

Refs
-  https://kubernetes.io/blog/2020/09/04/kubernetes-1-19-introducing-structured-logs/
-  https://github.com/kubernetes/klog

**Comment by damemi** [comment]
Yeah, this would be a good thing to work on. We've already done some work to switch our logs to structured in the descheduler too

**Comment by Huang-Wei** [comment]
@yuanchen8911 SGTM. But before starting, it'd be good to have a document listing basic principles to guide developers how and when to use structural logging. Moreover, if we want to transfer to structural logging entirely, we'd better come up with a CI script to guard that as well.

BTW: I recalled @damemi raised an issue in k/k to improve logging, right?

**Comment by damemi** [comment]
@Huang-Wei yep, that issue is here: https://github.com/kubernetes/kubernetes/issues/91633

**Comment by JaneLiuL** [comment]
Could I try to fix this issue as the good first issue that I can try to contribute to this gitrepository?
/assign

**Comment by pravarag** [comment]
I can take the `Part 1` for the logging issue as mentioned above :)

**Comment by arvidOtt** [comment]
I am working on part 2.

**Comment by CIPHERTron** [comment]
I'm willing to take up `Part 4` 🙌🏻

**Comment by sayantani11** [comment]
I'm willing to take part 3

**Comment by yharish991** [comment]
i can take part 5

**Comment by Darshnadas** [comment]
I am willing to take part 6

**Comment by Huang-Wei** [comment]
@Darshnadas Part 6 had an owner already :)

**Comment by Darshnadas** [comment]
Yes, I did not notice it earlier. No issues :)

**Comment by jyz0309** [comment]
I am willing to take part 5

**Comment by Huang-Wei** [comment]
> I am willing to take part 5. @Huang-Wei

it's claimed by @yharish991. @yharish991 are you still working part 5?

**Comment by jyz0309** [comment]
> > I am willing to take part 5. @Huang-Wei
> 
> it's claimed by @yharish991. @yharish991 are you still working part 5?

oh...I really sorry that I really have not noticed that. @yharish991  If you still work on this. I will close my PR.

**Comment by Huang-Wei** [comment]
> oh...I really sorry that I really have not noticed that.

That's no problem.

>  @yharish991 If you still work on this. I will close my PR.

Let's wait a couple of days (until this weekend). If he is not working on it, I will go review your PR.

**Comment by yharish991** [comment]
@jyz0309 @Huang-Wei i'm almost done with this work, and was about to open a PR, but since @jyz0309 already opened a PR, let's review his PR :) I'm good with that :)

**Comment by jyz0309** [comment]
> @jyz0309 @Huang-Wei i'm almost done with this work, and was about to open a PR, but since @jyz0309 already opened a PR, let's review his PR :) I'm good with that :)

No. Just open your PR. :) I have closed my PR.😉

**Comment by yharish991** [comment]
@Huang-Wei opened the pr for part 5 https://github.com/kubernetes-sigs/scheduler-plugins/pull/259

**Comment by pravarag** [comment]
I'll be working on the CI script part as mentioned here in this comment: https://github.com/kubernetes-sigs/scheduler-plugins/issues/120#issuecomment-737653251 .

**Comment by pravarag** [comment]
@Huang-Wei @damemi few questions on adding the CI check for structured logging:
1. The verify script will scan the diffs for the logging changes, if any. Is there a klog CLI which can be run during the job to verify the changes? Or is there any other way to filter for all the rules and structural changes in the commit.
2. I'm guessing the ideal place to put the verify script will be this location:https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/hack
3. For tracking the progress, shall I continue in this issue itself or open a new issue for the same?

Any other suggestions you have which I can work on, please let me know 🙂

**Comment by Huang-Wei** [comment]
Re 1: klog is more a SDK, so I don't think there is a CLI available. We don't need to get the changes in a commit. Instead, check the whole project. For sure, some folders like /vendor can be ignored. Ref: https://github.com/kubernetes/kubernetes/blob/master/hack/verify-gofmt.sh

Re 2: Nope, we should put it under `<project>/hack/`.

Re 3: Please open an new issue for tracking.

**Comment by pravarag** [comment]
I've created the following issue: https://github.com/kubernetes-sigs/scheduler-plugins/issues/261 for tracking further details/changes to it.

**Comment by Huang-Wei** [comment]
All items have been completed. Close this issue.
